### DISCUSSION TITLE
Issue after sending command to run certbot & dns-ionos

### DISCUSSION THREAD

**Discussion by alexeland** [helgeerbe/certbot-dns-ionos#18] [Q&A]
Hello,

Firstly I have to say that to arrive to this point I have battled with python, pip, certbot, etc... versioning to make certbot-dns-ionos work. In fact, the pypi doc (which is the same posted on this github repo's readme) candidly invites to install the package with **pip install**. 

Having never delt with python before, I installed it as suggested by apt: **apt install python3-pip**

Unfortunately, it installs pip with python 3.8 and I've witnessed that certbot-dns-ionos is not compatible with python 3.8. It threw an error while trying to run the certbot certonly command saying something about OpenSSL versioning (have not saved it unfortunately).

Trying multiple python versions and following the metadatas indicated on pypi, avoiding OpenSSL versioning errors depending on which python release is ran, I've resorted to manually install everything from python to certbot around python version 2.7.18 (so every package version is from around this release's date).

So far it worked, up until running the final certbot command:

`certbot certonly \
  --authenticator dns-ionos \
  --dns-ionos-credentials /etc/letsencrypt/.secrets/credentials.ini \
  --dns-ionos-propagation-seconds 900 \
  --installer nginx \
  --server https://acme-v02.api.letsencrypt.org/directory \
  --agree-tos \
  --rsa-key-size 4096 \
  -d ipfs.example.com -d ipns.example.com \
  -d *.ipfs.example.com -d *.ipns.example.com`

And it printed the error in the log file attached below. 
[log.txt](https://github.com/helgeerbe/certbot-dns-ionos/files/9894707/log.txt)

If it can help, while running **python setup.py install** the error shown in the log file printed while compiling:

`byte-compiling build/bdist.linux-x86_64/egg/certbot_dns_ionos/dns_ionos.py to dns_ionos.pyc
  File "build/bdist.linux-x86_64/egg/certbot_dns_ionos/dns_ionos.py", line 56
    logger.debug(f"_perform called with: domain: {domain}, validation_name: {validation_name}, validation: {validation}")
                                                                                                                       ^
SyntaxError: invalid syntax`

These two errors also showed up at the very beginning of running setup.py, though it makes no sense as they are related to not having setup ran by setuptools, however I have setuptools installed & it is imported at the beginning of your setup.py file.

`/usr/local/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'python_requires'
  warnings.warn(msg)
/usr/local/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'long_description_content_type'
  warnings.warn(msg)`

I have honestly no idea how to solve this, hence I'm coming here for help.

**Comment by helgeerbe** [comment]
Hi @alexeland,

certbot requires  Python >=3.7. 

Your error references to python 2.7.  So first of all, please assure that you have the proper python version running.

````sh
sudo apt update
sudo apt install python3
sudo apt install python3-pip
````

If you have multiple python versions ensure, that you use the right pip command.  So maybe pip ist not aliased correctly. 
````
pip3 install certbot
pip3 install certbot-dns-ionos
````
